### PR TITLE
New Job Ladder - update Engineer & Senior Engineer roles

### PR DESCRIPTION
**Pull request by slyyl**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,38 +1,74 @@
 # IWD dev team ladder
 
+This job ladder exposes all roles inside IWD development department, and define the goals, expectations and responsibilities of each role.
```
Comment (RETFU) [inline README.md]: IWD development department --> IWD dev team

I feel less big corp and more welcoming for new comer

File: README.md [hunk 2]
```diff
@@ -1,38 +1,74 @@
...
+Constant progression, and meeting your individual objectives will help you to climb it.
```
Comment (RETFU) [inline README.md]: To climb this ladder you know what to do 😄

File: README.md [hunk 3]
```diff
@@ -1,38 +1,74 @@
...
+
+Be aware that some basic expectations are not specified in the roles:
+* Being involved into team reviews
```
Comment (RETFU) [inline README.md]: You mean code review? Not clear

File: README.md [hunk 4]
```diff
@@ -1,38 +1,74 @@
...
+* Delivering software on-time
```
Comment (RETFU) [inline README.md]: For me that's part of a progression path. For E1 you have to learn how to deliver on time. For E2 it should be ok.

File: README.md [hunk 5]
```diff
@@ -1,38 +1,74 @@
...
+* Communicating, seeking help when stuck
+* Being a "team player"
+
 ## Developer roles
```
Comment (RETFU) [inline README.md]: If you want to be fancy, you can say "Individual contributor track or IC track" see https://medium.com/@adrian.chung/the-individual-contributor-track-vs-the-management-track-7b22cf85b749

File: README.md [hunk 6]
```diff
@@ -1,38 +1,74 @@
...
+### Senior Engineer I
+* Experience: ~4–7 years*
+* Report: Engineering Director
 * Goals: 
-  * Builds defined features, investigates and fixes bugs, writes tests
-  * Focus on growing as developer learning existing tools, resources and processes
-  * Communicates progress and when something is blocking
-  * Code review
-  * Respect team process
+  * Work on epics, with minimal guidance
```
Comment (RETFU) [inline README.md]: you should define epic at the end with a epics*

File: README.md [hunk 7]
```diff
@@ -1,38 +1,74 @@
...
+  * Work on features, without guidance
+  * Write tests, without guidance
+  * Review & Approve teammates work
+  * Master his project technical stack
```
Comment (RETFU) [inline README.md]: same for project

File: README.md [hunk 8]
```diff
@@ -1,38 +1,74 @@
...
+  * Expert in one technology
+  * Expert in one large area of the codebase 
+  * Guide Engineer I & II
```
Comment (RETFU) [inline README.md]: Guide --> Mentor

File: README.md [hunk 9]
```diff
@@ -1,38 +1,74 @@
...
-  * Makes others better through code reviews, documentation, technical guidance, and mentoring
-  * The best practice, process and documentation watchdog
+  * Work on epics, without guidance
+  * Work on features, without guidance
+  * Write tests, without guidance
+  * Review & Approve teammates work
+  * Master his project technical stack
+  * Expert in more than one technology
+  * Expert in many areas of the codebase 
+  * Guide Senior Engineer I, and Engineer I & II
```
Comment (RETFU) [inline README.md]: Mentor Senior Engineer I and Engineer II

If think you can let the focus of mentoring E1 for SE1

File: README.md [hunk 10]
```diff
@@ -1,38 +1,74 @@
...
+  * Take initiatives to find and qualify structural issues
+  * Helps managing product technical backlog
```
Comment (RETFU) [inline README.md]: Helps managing --> Manage technical backlog with LE

File: README.md [hunk 11]
```diff
@@ -1,38 +1,74 @@
...
+  * Knows, Defines, Challenges, Evangelizes best practices, processes and documentation
   * Develop domains/business skills
```
Comment (RETFU) [inline README.md]: E2: start learning domains/business skills
S1: develop domains/business skills
S2: expert domains/business skills

File: README.md [hunk 12]
```diff
@@ -1,38 +1,74 @@
...
 
-### Level 3: Lead developer
+### Lead Engineer
 * Experience: ~10+ years*
-* Report: CTO
+* Report: Engineering Director
 * Goals: 
   * Lead a product team
```
Comment (RETFU) [inline README.md]: define a product team

File: README.md [hunk 13]
```diff
@@ -1,38 +1,74 @@
...
   * Owns the development and rollout for the entire product
-  * Promote culture, process, best practice
+  * Knows, Defines, Challenges, Evangelizes best practices, processes and culture
   * Identifies risks before starting major projects
   * Listen and guide debates to help reach a consensus
   * Goes out of their way to reduce complexity
```
Comment (RETFU) [inline README.md]: * Master his project technical stack
* Expert in more than one technology
* Expert in many areas of the codebase
* Mastering domains/business

**Comment by RETFU** [comment]
Also you can include Ops role in this IC track

In manager track 
* Engineer director TODO
* CTO to TODO

For a new comer that will give a clear picture about the organization and the fact that for now their is no management track available to transition, (ex: engineer manager...)

File: README.md (hunk 5 above)
Comment (slyyl) [reply README.md]: The way I understand this article is that the IC track is the ladder an IC can follow.

Being a team player (i.e. helping others, communicating, understanding/following established rules, ...) is a different point to me.

File: README.md (hunk 1 above)
Comment (slyyl) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (slyyl) [reply README.md]: Fixed

File: README.md (hunk 3 above)
Comment (slyyl) [reply README.md]: Fixed

File: README.md (hunk 4 above)
Comment (slyyl) [reply README.md]: Fixed

File: README.md (hunk 9 above)
Comment (slyyl) [reply README.md]: fixed

File: README.md (hunk 10 above)
Comment (slyyl) [reply README.md]: fixed

File: README.md (hunk 11 above)
Comment (slyyl) [reply README.md]: fixed

File: README.md (hunk 5 above)
Comment (RETFU) [reply README.md]: I'm talking about Developer roles --> Individual Contributor

File: README.md [hunk 14]
```diff
@@ -1,78 +1,151 @@
...
+- Goals:
+  - Lead a Squad team (*see definition above*)
+    - Supervise project's development, deadlines, rollout & support
+    - Manage technical backlog
+    - Report Squad/project status to Engineering Director
+    - Involved in hiring process
+    - Mentor Senior Engineer I & II
+    - Listen and guide debates to help the squad reaching a consensus
+    - Product Owner main contact
+  - Project's technical stack guarantor
```
Comment (RETFU) [inline README.md]: seems not super english -> Stack owner

File: README.md [hunk 15]
```diff
@@ -1,78 +1,151 @@
...
+  - Master his project technical stack
+  - Expert in more than one technology
+  - Expert in many areas of the codebase
+  - Mentor Engineer II & Senior Engineer I
+  - Take initiatives to find and qualify structural issues
+  - Manage technical backlog with Lead Engineer
+  - Know, Define, Challenge, Evangelize best practices, processes and documentation
+  - Expert in domain / business skills
+
+### Lead Engineer
```
Comment (RETFU) [inline README.md:79]: You should also introduce the transversality of the role, maybe: Can manage cross-team project

**Comment by RETFU** [comment]
Sound good to me

**Pull request merged**